src/backend/scripts/run-with-gcp-metadata.ts

```diff
@@ -0,0 +1,63 @@
+import { execa } from "execa";
+import * as gcpMetadata from "gcp-metadata";
+import axios from "axios";
+
+/**
+ * This script injects Google Cloud metadata (when available) into the environment.
+ * Specifically, it injects the NEXTAUTH_URL and BASE_HTTP_ENDPOINT environment variables.
+ * Run with `npm run with-gcp-metadata <command> <args>`.
+ */
```

Comment:
very frustrating: the generated cloud run URL isn't known until after the deploy, and terraform blocks can't reference themselves

I think this is the cleanest solution, didn't see anything simpler online

Reply:
NEXAUTH_URL will be our domain, e.g. [staging.]compassiep.com

Reply:
yes, however ideally we can deploy without a domain/with the generated one with necessary
it'd be nice to be able to deploy to a new project or a personal account for testing